### PR TITLE
[TACHYON-1146] Rename TachyonStorageType to NativeStorageType

### PR DESCRIPTION
**Pull request by calvinjia**
https://tachyon.atlassian.net/browse/TACHYON-1146

### CODE REVIEW

File: docs/Configuration-Settings.md [hunk 1]
```diff
@@ -556,14 +556,15 @@ The user configuration specifies values regarding file system access.
...
   <td>The default interaction with Tachyon. Possible values are PROMOTE, STORE, and NO_STORE.
-  Store will attempt to write data to Tachyon if the local worker does not have the data. This
-  applies to writing new data as well as reading data which is not already on the local worker.
-  Promote behaves the same as store, except if the data is on the local worker, promote will
-  migrate the data to the highest tier. No store will never attempt to write data into Tachyon
-  storage. This is useful to prevent one-time data access from interfering with Tachyon.</td>
+  Store will attempt to write data to Tachyon's natively managed storage if the local worker does
+  not have the data. This applies to writing new data as well as reading data which is not already
+  on the local worker. Promote behaves the same as store, except if the data is on the local worker,
+  promote will migrate the data to the highest tier. No store will never attempt to write data into
```
Comment (jsimsa) [inline docs/Configuration-Settings.md]: s/never/ever

Comment (jsimsa) [inline docs/Configuration-Settings.md]: s/No store/NO_STORE

File: docs/Configuration-Settings.md [hunk 2]
```diff
@@ -556,14 +556,15 @@ The user configuration specifies values regarding file system access.
...
   <td>STORE</td>
   <td>The default interaction with Tachyon. Possible values are PROMOTE, STORE, and NO_STORE.
-  Store will attempt to write data to Tachyon if the local worker does not have the data. This
-  applies to writing new data as well as reading data which is not already on the local worker.
-  Promote behaves the same as store, except if the data is on the local worker, promote will
-  migrate the data to the highest tier. No store will never attempt to write data into Tachyon
-  storage. This is useful to prevent one-time data access from interfering with Tachyon.</td>
+  Store will attempt to write data to Tachyon's natively managed storage if the local worker does
+  not have the data. This applies to writing new data as well as reading data which is not already
+  on the local worker. Promote behaves the same as store, except if the data is on the local worker,
```
Comment (jsimsa) [inline docs/Configuration-Settings.md]: s/Promote/PROMOTE

File: docs/Configuration-Settings.md [hunk 3]
```diff
@@ -556,14 +556,15 @@ The user configuration specifies values regarding file system access.
...
-  <td>tachyon.user.file.tachyonstoragetype.default</td>
+  <td>tachyon.user.file.nativestoragetype.default</td>
   <td>STORE</td>
   <td>The default interaction with Tachyon. Possible values are PROMOTE, STORE, and NO_STORE.
-  Store will attempt to write data to Tachyon if the local worker does not have the data. This
-  applies to writing new data as well as reading data which is not already on the local worker.
-  Promote behaves the same as store, except if the data is on the local worker, promote will
-  migrate the data to the highest tier. No store will never attempt to write data into Tachyon
-  storage. This is useful to prevent one-time data access from interfering with Tachyon.</td>
+  Store will attempt to write data to Tachyon's natively managed storage if the local worker does
```
Comment (jsimsa) [inline docs/Configuration-Settings.md]: s/Store/STORE

File: docs/Configuration-Settings.md [hunk 4]
```diff
@@ -556,14 +556,15 @@ The user configuration specifies values regarding file system access.
...
+  not have the data. This applies to writing new data as well as reading data which is not already
+  on the local worker. Promote behaves the same as store, except if the data is on the local worker,
+  promote will migrate the data to the highest tier. No store will never attempt to write data into
+  Tachyon's managed storage. This is useful to prevent one-time data access from interfering with
```
Comment (jsimsa) [inline docs/Configuration-Settings.md]: s/This/The latter

File: docs/File-System-API.md [hunk 5]
```diff
@@ -30,7 +30,7 @@ TachyonFileSystem tfs = TachyonFileSystemFactory.get();
 ### Creating a File
 
 All metadata operations as well as opening a file for reading or creating a file for writing are
-done through the TachyonFileSystem object. Since Tachyon files are immutable once written, the 
+done through the TachyonFileSystem object. Since Tachyon files are immutable once written, the
```
Comment (jsimsa) [inline docs/File-System-API.md]: s/done/carried out

File: docs/File-System-API.md [hunk 6]
```diff
@@ -58,14 +58,69 @@ OutStreamOptions options = new OutStreamOptions.Builder(ClientContext.getConf())
 FileOutStream out = tfs.getOutputStream(path, options);
 ```
 
+### IO Options
+
+There are two different types of storage in Tachyon, natively managed storage and under storage.
```
Comment (jsimsa) [inline docs/File-System-API.md]: Tachyon uses two different storage types: natively managed storage and under storage.

File: docs/File-System-API.md [hunk 7]
```diff
@@ -58,14 +58,69 @@ OutStreamOptions options = new OutStreamOptions.Builder(ClientContext.getConf())
...
+Natively managed storage is the memory, SSD, and/or HDD which is allocated to Tachyon workers. For
```
Comment (jsimsa) [inline docs/File-System-API.md]: s/which is allocated to/allocated for

File: docs/File-System-API.md [hunk 8]
```diff
@@ -58,14 +58,69 @@ OutStreamOptions options = new OutStreamOptions.Builder(ClientContext.getConf())
...
+read and write operations, users can specify the `NativeStorageType` to fine tune the interaction
```
Comment (jsimsa) [inline docs/File-System-API.md]: s/fine tune/specify

Comment (jsimsa) [inline docs/File-System-API.md]: users can specify the interaction with Tachyon's native storage through the `NativeStorageType`.

File: docs/Configuration-Settings.md (hunk 1 above)
Comment (calvinjia) [inline docs/Configuration-Settings.md]: Updated, thanks.

File: docs/Configuration-Settings.md (hunk 2 above)
Comment (calvinjia) [inline docs/Configuration-Settings.md]: Updated, thanks.

File: docs/Configuration-Settings.md (hunk 3 above)
Comment (calvinjia) [inline docs/Configuration-Settings.md]: Updated, thanks.

File: docs/Configuration-Settings.md (hunk 4 above)
Comment (calvinjia) [inline docs/Configuration-Settings.md]: Updated, thanks.

File: docs/File-System-API.md (hunk 5 above)
Comment (calvinjia) [inline docs/File-System-API.md]: Changed to executed.

File: docs/File-System-API.md (hunk 6 above)
Comment (calvinjia) [inline docs/File-System-API.md]: Updated, thanks.

File: docs/File-System-API.md (hunk 7 above)
Comment (calvinjia) [inline docs/File-System-API.md]: Updated, thanks.

File: docs/File-System-API.md (hunk 8 above)
Comment (calvinjia) [inline docs/File-System-API.md]: Updated, thanks.

**Pull request closed without merging**